### PR TITLE
feat(onesync/helicopters): serialize helicopters datanodes

### PR DESCRIPTION
**Pull request by ahcenezdh**
### Goal of this PR
* New getters :
     - GET_THRUSTER_THROTTLE
     - GET_THRUSTER_SIDE_RCS_THROTTLE
     - IS_HELI_TAIL_BOOM_BROKEN
     - GET_IS_HELI_ENGINE_RUNNING
     - GET_HELI_YAW_CONTROL
     - GET_HELI_THROTTLE_CONTROL
     - GET_HELI_TAIL_ROTOR_DAMAGE_SCALE
     - GET_HELI_ROLL_CONTROL
     - GET_HELI_REAR_ROTOR_HEALTH
     - GET_HELI_REAR_ROTOR_DAMAGE_SCALE
     - GET_HELI_PITCH_CONTROL
     - GET_HELI_MAIN_ROTOR_DAMAGE_SCALE
     - GET_HELI_GAS_TANK_HEALTH
     -  GET_HELI_ENGINE_HEALTH
     - GET_HELI_DISABLE_EXPLODE_FROM_BODY_DAMAGE
     - GET_HELI_BODY_HEALTH
     - IS_HELI_TAIL_BOOM_BREAKABLE

`GET_HELI_TAIL_ROTOR_HEALTH` was incorrect, it was reading the rear rotor bit instead of the tail rotor (which doesn't exist). To address this, I created a new native `GET_HELI_REAR_ROTOR_HEALTH` and added a deprecation warning to GET_HELI_TAIL_ROTOR_HEALTH using ScriptDeprecations, as requested by @FabianTerhorst

### How is this PR achieving the goal
By serializing `CHeliHealthDataNode` and `CHeliControlDataNode` and exposing new getters

### This PR applies to the following area(s)

FiveM, FXServer
...

### Successfully tested on

**Game builds:** 3095 

**Platforms:** Windows

### Checklist

- [x] Code compiles and has been tested successfully.
- [x] Code explains itself well and/or is documented.
- [x] My commit message explains what the changes do and what they are for.
- [x] No extra compilation warnings are added by these changes.

### CODE REVIEW

**Comment by ahcenezdh** [comment]
Here is a command to test getters from the health data node
```lua
RegisterCommand("heliHealth", function(source)
    local playerPed = GetPlayerPed(source)
    local veh = GetVehiclePedIsIn(playerPed)
    local heliMainRotorHealth = GetHeliMainRotorHealth(veh)
    local heliTailRotorHealth = GetHeliTailRotorHealth(veh)
    local heliRearRotorHealth = GetHeliRearRotorHealth(veh)

    local isTailBoomBroken = GetIsHeliTailBoomBroken(veh)
    local canTailBoomBreak = CanHeliTailBoomBreak(veh)

    local getHeliBodyHealth = GetHeliBodyHealth(veh)
    local gasTank = GetHeliGasTankHealth(veh)

    local heliEngine = GetHeliEngineHealth(veh)
    local mainRotorDamage = GetHeliMainRotorDamageScale(veh)
    local rearRotorDamage = GetHeliRearRotorDamageScale(veh)
    local tailRotorDamage = GetHeliTailRotorDamageScale(veh)
    local disableExplodeFromBodyDamage = GetHeliDisableExplodeFromBodyDamage(veh)

    print(string.format("Heli Main Rotor Health: %s", heliMainRotorHealth))
    print(string.format("Heli Tail Rotor Health: %s", heliTailRotorHealth))
    print(string.format("Heli Rear Rotor Health: %s", heliRearRotorHealth))

    print(string.format("Is Tail Boom Broken: %s", isTailBoomBroken))
    print(string.format("Can Tail Boom Break: %s", canTailBoomBreak))

    print(string.format("Heli Body Health: %s", getHeliBodyHealth))
    print(string.format("Heli Gas Tank Health: %s", gasTank))

    print(string.format("Heli Engine Health: %s", heliEngine))
    print(string.format("Main Rotor Damage: %s", mainRotorDamage))
    print(string.format("Rear Rotor Damage: %s", rearRotorDamage))
    print(string.format("Tail Rotor Damage: %s", tailRotorDamage))
    print(string.format("Disable Explode From Body Damage: %s", disableExplodeFromBodyDamage))
end, false)
```
You can play with client side setters to change some values, please refer to docs provided with natives

**Comment by ahcenezdh** [comment]
Here is another commands to test natives from the control data node.

```lua
RegisterCommand("heliControl", function()
    local playerPed = GetPlayerPed(source)
    local playerVeh = GetVehiclePedIsIn(playerPed, false)

    -- We are printing those values every 5 seconds since they change when the helicopter moves.

    CreateThread(function()
        while (true) do
            Wait(5000)
            local yawControl = GetHeliYawControl(playerVeh)
            local pitchControl = GetHeliPitchControl(playerVeh)
            local rollControl = GetHeliRollControl(playerVeh)
            local throttleControl = GetHeliThrottleControl(playerVeh)

            if (yawControl ~= 0.0) then
                print("Yaw Control", yawControl)        
            end
        
            if (pitchControl ~= 0.0) then
                print("Pitch Control", pitchControl)        
            end
        
            if (rollControl ~= 0.0) then
                print("Roll Control", rollControl) 
            end
        
            if (throttleControl ~= 0.0) then
                print("Throttle Control", throttleControl)
            end
        end
    end)

    -- Jetpack Natives

    local jetpackEnable = false

    CreateThread(function()
        while jetpackEnable do
            Wait(5000)

            local RCS = GetThrusterSideRcsThrottle(playerVeh)
            local thrusterThrottle = GetThrusterThrottle(playerVeh)
        
            if RCS ~= 0.0 then
                print("Thruster RCS", RCS) 
            end
        
            if (thrusterThrottle ~= 0.0) then
                print("Thruster Throttle", thrusterThrottle)            
            end
        end
    end)
end)
```

**Review by PsychoShock — COMMENTED** [review]
You could also add for both ([CFX ](https://github.com/citizenfx/fivem/edit/master/ext/native-decls/GetHeliTailRotorHealth.md)and [Vehicle](https://github.com/citizenfx/natives/edit/master/VEHICLE/GetHeliTailRotorHealth.md)) of `GetHeliTailRotorHealth` the decraption description to prevent players using it wrong.

**Comment by ahcenezdh** [comment]
CanHeliTailBoomBreak if someone have a better name for this native?

**Comment by PsychoShock** [comment]
> CanHeliTailBoomBreak if someone have a better name for this native?

R* use a lot "IS_XXX" so maybe `IS_HELI_TAIL_BOOM_BREAKABLE`.

**Comment by ahcenezdh** [comment]
> > CanHeliTailBoomBreak if someone have a better name for this native?
> 
> R* use a lot "IS_XXX" so maybe `IS_HELI_TAIL_BOOM_BREAKABLE`.

@FabianTerhorst Is it good for you?

**Comment by FabianTerhorst** [comment]
> > > CanHeliTailBoomBreak if someone have a better name for this native?
> > 
> > 
> > R* use a lot "IS_XXX" so maybe `IS_HELI_TAIL_BOOM_BREAKABLE`.
> 
> @FabianTerhorst Is it good for you?

IS_HELI_TAIL_BOOM_BREAKABLE sounds good yes. It is in general better to start the bool returning natives with `is` to better indicate that they return booleans when the name allows it.

**Comment by PsychoShock** [comment]
Imo `GET_IS_HELI_TAIL_BOOM_BROKEN` should just be `IS_HELI_TAIL_BOOM_BROKEN`. Just sound better/shorter.

It will also match [other heli natives](https://docs.fivem.net/natives/?_0xBC74B4BE25EB6C8A) name.

**Comment by ahcenezdh** [comment]
I did a PR to updates natives docs.

**Comment by ahcenezdh** [comment]
> Imo `GET_IS_HELI_TAIL_BOOM_BROKEN` should just be `IS_HELI_TAIL_BOOM_BROKEN`. Just sound better/shorter.
> 
> It will also match [other heli natives](https://docs.fivem.net/natives/?_0xBC74B4BE25EB6C8A) name.

Yes ur right, should be good

**Review by FabianTerhorst — APPROVED** [review]
Thanks for your contribution. Everything works fine for me. 👍

File: ext/native-decls/GetHeliBodyHealth.md [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+apiset: server
+game: gta5
+---
+## GET_HELI_BODY_HEALTH
+
+```c
+int GET_HELI_BODY_HEALTH(Vehicle heli);
+```
+
+**Note**: If any of the following natives: [SET_VEHICLE_BODY_HEALTH](https://docs.fivem.net/natives/?_0xB77D05AC8C78AADB), [SET_VEHICLE_ENGINE_HEALTH](https://docs.fivem.net/natives/?_0x45F6D8EEF34ABEF1), or [SET_VEHICLE_PETROL_TANK_HEALTH](https://docs.fivem.net/natives/?_0x70DB57649FA8D0D8) is used with a value greater than `1000`, the health data for the helicopter becomes accessible through all related getter natives, even if their specific values are set below `1000`. Setting any one of these natives above `1000` allows the others to be used with any value.
```
Comment (AvarianKnight) [inline ext/native-decls/GetHeliBodyHealth.md:12]:
```suggestion
**Note** This native will always return `1000.0` unless [SET_VEHICLE_BODY_HEALTH](#_0xB77D05AC8C78AADB), [SET_VEHICLE_ENGINE_HEALTH](#_0x45F6D8EEF34ABEF1), or [SET_VEHICLE_PETROL_TANK_HEALTH](#_0x70DB57649FA8D0D8) have been called with a value greater than `1000.0`.
```
This should simplify the wording and hopefully make it easier to understand. This should probably also be put on the natives repo as a disclaimer for these natives too.

File: ext/native-decls/GetHeliBodyHealth.md [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+
+## Parameters
+* **heli**: The helicopter to check
+
+## Return value
+Returns the current health of the helicopter's body. By default, this native returns `1000` unless [SET_VEHICLE_BODY_HEALTH](https://docs.fivem.net/natives/?_0xB77D05AC8C78AADB) has been used with a value greater than `1000`.
```
Comment (AvarianKnight) [inline ext/native-decls/GetHeliBodyHealth.md:18]:
```suggestion
Returns the current health of the helicopter's body.
```

Remove duplicate disclaimer

File: ext/native-decls/GetHeliDisableExplodeFromBodyDamage.md [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+apiset: server
+game: gta5
+---
+## GET_HELI_DISABLE_EXPLODE_FROM_BODY_DAMAGE
+
+```c
+BOOL GET_HELI_DISABLE_EXPLODE_FROM_BODY_DAMAGE(Vehicle heli);
+```
+
+**Note**: This is a getter for [SET_DISABLE_HELI_EXPLODE_FROM_BODY_DAMAGE](https://docs.fivem.net/natives/?_0xEDBC8405B3895CC9)
```
Comment (AvarianKnight) [inline ext/native-decls/GetHeliDisableExplodeFromBodyDamage.md:12]:
```suggestion
This is a getter for [SET_DISABLE_HELI_EXPLODE_FROM_BODY_DAMAGE](#_0xEDBC8405B3895CC9)
```

File: ext/native-decls/GetHeliEngineHealth.md [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+apiset: server
+game: gta5
+---
+## GET_HELI_ENGINE_HEALTH
+
+```c
+int GET_HELI_ENGINE_HEALTH(Vehicle heli);
+```
+
+**Note**: If any of the following natives: [SET_VEHICLE_BODY_HEALTH](https://docs.fivem.net/natives/?_0xB77D05AC8C78AADB), [SET_VEHICLE_ENGINE_HEALTH](https://docs.fivem.net/natives/?_0x45F6D8EEF34ABEF1), or [SET_VEHICLE_PETROL_TANK_HEALTH](https://docs.fivem.net/natives/?_0x70DB57649FA8D0D8) is used with a value greater than `1000`, the health data for the helicopter becomes accessible through all related getter natives, even if their specific values are set below `1000`. Setting any one of these natives above `1000` allows the others to be used with any value.
```
Comment (AvarianKnight) [inline ext/native-decls/GetHeliEngineHealth.md:12]:
```suggestion
**Note** This native will always return `1000.0` unless [SET_VEHICLE_BODY_HEALTH](#_0xB77D05AC8C78AADB), [SET_VEHICLE_ENGINE_HEALTH](#_0x45F6D8EEF34ABEF1), or [SET_VEHICLE_PETROL_TANK_HEALTH](#_0x70DB57649FA8D0D8) have been called with a value greater than `1000.0`.
```

File: ext/native-decls/GetHeliEngineHealth.md [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+
+## Parameters
+* **heli**: The helicopter to check
+
+## Return value
+Returns the current health of the helicopter's engine. By default, this native returns `1000` unless [SET_VEHICLE_BODY_HEALTH](https://docs.fivem.net/natives/?_0xB77D05AC8C78AADB) has been used with a value greater than `1000`.
```
Comment (AvarianKnight) [inline ext/native-decls/GetHeliEngineHealth.md:18]:
```suggestion
Returns the current health of the helicopter's engine.
```

File: ext/native-decls/GetHeliGasTankHealth.md [hunk 6]
```diff
@@ -0,0 +1,18 @@
...
+apiset: server
+game: gta5
+---
+## GET_HELI_GAS_TANK_HEALTH
+
+```c
+int GET_HELI_GAS_TANK_HEALTH(Vehicle heli);
+```
+
+**Note**: If any of the following natives: [SET_VEHICLE_BODY_HEALTH](https://docs.fivem.net/natives/?_0xB77D05AC8C78AADB), [SET_VEHICLE_ENGINE_HEALTH](https://docs.fivem.net/natives/?_0x45F6D8EEF34ABEF1), or [SET_VEHICLE_PETROL_TANK_HEALTH](https://docs.fivem.net/natives/?_0x70DB57649FA8D0D8) is used with a value greater than `1000`, the health data for the helicopter becomes accessible through all related getter natives, even if their specific values are set below `1000`. Setting any one of these natives above `1000` allows the others to be used with any value.
```
Comment (AvarianKnight) [inline ext/native-decls/GetHeliGasTankHealth.md:12]:
```suggestion
**Note** This native will always return `1000.0` unless [SET_VEHICLE_BODY_HEALTH](#_0xB77D05AC8C78AADB), [SET_VEHICLE_ENGINE_HEALTH](#_0x45F6D8EEF34ABEF1), or [SET_VEHICLE_PETROL_TANK_HEALTH](#_0x70DB57649FA8D0D8) have been called with a value greater than `1000.0`.
```

File: ext/native-decls/GetHeliGasTankHealth.md [hunk 7]
```diff
@@ -0,0 +1,18 @@
...
+
+## Parameters
+* **heli**: The helicopter to check
+
+## Return value
+Returns the current health of the helicopter's gas tank. By default, this native returns `1000` unless [SET_VEHICLE_PETROL_TANK_HEALTH](https://docs.fivem.net/natives/?_0x70DB57649FA8D0D8) has been used with a value greater than the actual petrol tank health of the helicopter.
```
Comment (AvarianKnight) [inline ext/native-decls/GetHeliGasTankHealth.md:18]:
```suggestion
Returns the current health of the helicopter's gas tank.
```

File: ext/native-decls/GetHeliRearRotorHealth.md [hunk 8]
```diff
@@ -0,0 +1,18 @@
...
+apiset: server
+game: gta5
+---
+## GET_HELI_REAR_ROTOR_HEALTH
+
+```c
+float GET_HELI_REAR_ROTOR_HEALTH(Vehicle vehicle);
+```
+
+**Note**: This native is a getter for [GET_HELI_TAIL_ROTOR_HEALTH](https://docs.fivem.net/natives/?_0xAE8CE82A4219AC8C)
```
Comment (AvarianKnight) [inline ext/native-decls/GetHeliRearRotorHealth.md:12]:
```suggestion
This native is a getter for [SET_HELI_TAIL_ROTOR_HEALTH](#_0xFE205F38AAA58E5B)

```

File: ext/native-decls/GetHeliRearRotorHealth.md [hunk 9]
```diff
@@ -0,0 +1,18 @@
...
+
+## Parameters
+* **vehicle**: The target vehicle.
+
+## Return value
+Returns the health of the helicopter's rear rotor. The maximum health value is `1000`.
```
Comment (AvarianKnight) [inline ext/native-decls/GetHeliRearRotorHealth.md:18]: Has it been verified that this actually clamps the max health?

File: ext/native-decls/GetHeliTailRotorHealth.md [hunk 10]
```diff
@@ -8,9 +9,10 @@ apiset: server
 float GET_HELI_TAIL_ROTOR_HEALTH(Vehicle vehicle);
 ```
 
+**Note**: This native is deprecated, please use `GET_HELI_REAR_ROTOR_HEALTH` instead.
```
Comment (AvarianKnight) [inline ext/native-decls/GetHeliTailRotorHealth.md:12]:
```suggestion
**Note**: This native is deprecated, please use [`GET_HELI_REAR_ROTOR_HEALTH`](#_0x33EE6E2B) instead.
```

Comment (ahcenezdh) [reply ext/native-decls/GetHeliTailRotorHealth.md:12]: Oh i was searching how to do it, i don't understand what is 0x33EE6E2B? it's the joaat of the native?

Comment (AvarianKnight) [reply ext/native-decls/GetHeliTailRotorHealth.md:12]: Yes all Cfx native will be a jooat.

File: ext/native-decls/GetHeliBodyHealth.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (ahcenezdh) [reply ext/native-decls/GetHeliBodyHealth.md:12]: I did a PR to update some natives names used in this PR, but yeah some notes should be added on those setters

File: ext/native-decls/GetHeliRearRotorHealth.md (hunk 9 above)
Comment (ahcenezdh) [reply ext/native-decls/GetHeliRearRotorHealth.md:18]: Yes, but I may be wrong, feel free to double-check
